### ISSUE TITLE
Puzzle mod has duplicate buttons

### ISSUE THREAD

**Issue by magicus** [magicus/ModSettings#12]
Originally reported by @ardissaps in #10:

> Puzzle mod have 3 button on the setting menu which is if pressed open same screen (duplicate/redundant button).

**Comment by magicus** [comment]
The problem with Puzzle mod is most likely with the mod itself. My guess is that it is bundling separate mods which all have the same configuration screen.

Is it this mod: https://www.curseforge.com/minecraft/mc-mods/puzzle ?

**Comment by magicus** [comment]
Note to self: A work-around might be to look for duplicate mod titles, and/or config screen provider classes. It might also be worthwhile to try and hide configuration for mods that are imported indirectly from other mods.

**Comment by ardissaps** [comment]
> Is it this mod: https://www.curseforge.com/minecraft/mc-mods/puzzle ?

Yes it is.

**Comment by magicus** [comment]
@mrniddala This is not the same issue. Please open a separate bug report.

**Comment by magicus** [comment]
Puzzle mod is here on Github: https://github.com/PuzzleMC/Puzzle

**Comment by magicus** [comment]
This was actually a sign of a deeper problem. Some mods bundle other, in a parent-child relationship. Most of the time, these bundled mods are library mods with no settings screen, but sometimes they are. For instance, ClothConfig adds a demo screen that is not really helpful.

I have now added functionality in 2.0.0 to hide such child mods by default (but still being able to access them if needed). I have verified that there is now only a single Puzzle button.
